sim/dex-formats.ts

```diff
@@ -596,11 +611,23 @@ export class DexFormats {
 				ruleTable.minSourceGen = subRuleTable.minSourceGen;
 			}
 		}
+		this.getTagRules(ruleTable);
 
 		if (!repeals) format.ruleTable = ruleTable;
 		return ruleTable;
 	}
 
+	getTagRules(ruleTable: RuleTable) {
+		const tagRules = [];
+		for (const ruleid of ruleTable.keys()) {
+			if (ruleid.startsWith('+pokemontag:') || ruleid.startsWith('-pokemontag:') || ruleid.startsWith('*pokemontag:')) {
```

Comment:
```suggestion
			if (/^[+-*]pokemontag:/.test(ruleid)) {
```
Is there a reason to not use regex here? I don't know if so.

Reply:
It has to be `[+*-]` - `-` means something different when it's in the middle of a range. But otherwise yes, you're right, regex is better.

Reply:
Oh fair, I just kinda typed swiftly

Reply:
```suggestion
			if (/^[+*-]pokemontag:/.test(ruleid)) {
```